cirq-google/cirq_google/study/var.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2023 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, Iterable, Optional, Sequence, Union
+import dataclasses
+from typing_extensions import Protocol
+
+import sympy
+
+import cirq
+
+
+class SupportsParameter(Protocol):
+    """Protocol for using non-circuit parameter keys.
+
+    For instance, verying the length of pulses, timing, etc.
+
+    Args:
+       path: path of the key to modify, with each sub-folder as a string
+           entry in a list.
+       idx: If this key is an array, which index to modify.
+    """
+
+    path: Sequence[str]
+    idx: Optional[int] = None
+    value: Optional[Any] = None
+
+
+@dataclasses.dataclass
+class Parameter:
+    """Reference implementation of a generic parameter."""
+
+    path: Sequence[str]
+    idx: Optional[int] = None
+    value: Optional[Any] = None
+
+    @classmethod
+    def _json_namespace_(cls) -> str:
+        return 'cirq.google'
+
+    @classmethod
+    def _from_json_dict_(cls, path, idx, value, **kwargs):
+        return Parameter(path=path, idx=idx, value=value)
+
+    def _json_dict_(self) -> Dict[str, Any]:
+        return cirq.obj_to_dict_helper(self, ["path", "idx", "value"])
+
+
+class Var(cirq.Points):
+    """Class for representing a non-Symbol variable.
+
+    Internally, certain parameters are not represented as
```

Comment:
Nit: remove "internally" since that will now be true for the external context as well.